Source/WebGPU/WGSL/AST/ShaderModule.h

```diff
@@ -39,41 +39,43 @@ namespace WGSL::AST {
 
 class ShaderModule final : ASTNode {
     WTF_MAKE_FAST_ALLOCATED;
+
 public:
-    ShaderModule(SourceSpan span, Vector<UniqueRef<GlobalDirective>>&& directives, Vector<UniqueRef<Decl>>&& decls)
+    ShaderModule(SourceSpan span, GlobalDirective::List&& directives, Decl::List&& decls)
         : ASTNode(span)
         , m_directives(WTFMove(directives))
     {
-        for (auto& decl : decls) {
-            if (is<StructDecl>(decl.get())) {
+        for (size_t i = decls.size(); i > 0; --i) {
+            UniqueRef<Decl> decl = decls.takeLast();
+            if (is<StructDecl>(decl)) {
```

Comment:
Any reason you changed this?

Reply:
Because the UniqueRefVector change means that decl is now Decl& not UniqueRef<Decl>& and Decl is move only, I’m taking UniqueRef out of the vector and downcasting them. This is going away in a follow up.